### DISCUSSION TITLE
What should be the X-Session-Token header value?

### DISCUSSION THREAD

**Discussion by trobinpl** [ory/kratos#1415] [Q&A] [answered]
In my application I would like to hide the Kratos calls behind my own set of APIs. Documentation says that in such case the X-Session-Token header should be set, but I'm not sure what's supposed to be it's value. When I issue a request to `session/whoami` with the `ory_kratos_session` cookie in place, I get the desired response. However if I try to put the same value in the X-Session-Token header (or Bearer for that matter) I get 401.

I was under the impression that you can substitute cookie with appropriate header, but it seems not to be the case. What's the correct approach then?

**Comment by aeneasr** [comment, marked as answer]
This is definitely unclear in the docs. We’ll be adding a dimple way to forward the cookie to the whoami endpoint in the open SPA PR!

To solve it right now you need to forward the incoming HTTP headers to the whoami endpoint :)

**Reply by trobinpl** [reply]
Yep, that's what I ended up doing and it did indeed work 😃

**Reply by krhubert** [reply]
@aeneasr Is it possible to pass incoming HTTP headers in `kratos-client-go`? There's only method to pass `X-Session-Token`.

**Reply by aeneasr** [reply]
Yeah 0.7 has a much improved way for this!

**Reply by aeneasr** [reply]
Coming out probably next week

**Reply by krhubert** [reply]
Ok found https://github.com/ory/kratos/discussions/1398 here as well, thanks!
